src/protocol/present.rs

```diff
@@ -588,6 +589,10 @@ impl TryParse for QueryVersionReply {
             return Err(ParseError::ParseError);
         }
         let result = QueryVersionReply { sequence, length, major_version, minor_version };
+        let remaining = remaining.get((4 - (initial_value.len() - remaining.len()) % 4) % 4..)
+            .ok_or(ParseError::ParseError)?;
+        let remaining = remaining.get(32usize.saturating_sub(initial_value.len() - remaining.len())..)
+            .ok_or(ParseError::ParseError)?;
```

Comment:
Hm. Okay... for GE events, my proposed code does not work. Perhaps it makes sense to tell apart "this is always 32 bytes"-types and "this has a length field called `length`" in the code generator and do something based on that?

Does that sound sensible?

(Also, I really should stop doing single comments on GH, even when I am *sure* that I will only write one comment. That never works out...)